keystore/src/transaction.rs

```diff
@@ -0,0 +1,340 @@
+use std::{collections::VecDeque, sync::Arc};
+
+use async_lock::RwLock;
+use openmls_basic_credential::SignatureKeyPair;
+use openmls_traits::key_store::OpenMlsKeyStore;
+use openmls_traits::key_store::{MlsEntity, MlsEntityId};
+
+use crate::{
+    connection::{Connection, DatabaseConnection, TransactionWrapper},
+    entities::{
+        E2eiAcmeCA, E2eiCrl, E2eiEnrollment, E2eiIntermediateCert, E2eiRefreshToken, EntityMlsExt, MlsCredential,
+        MlsEncryptionKeyPair, MlsEpochEncryptionKeyPair, MlsHpkePrivateKey, MlsKeyPackage, MlsPendingMessage,
+        MlsPskBundle, MlsSignatureKeyPair, PersistedMlsGroup, PersistedMlsPendingGroup, StringEntityId,
+    },
+    CryptoKeystoreError, CryptoKeystoreResult,
+};
+
```

Comment:
Both the below enums were created to avoid having to rewrite and re-design the whole entity structure. There are several very strict restrictions imposed by the `EntityBase` trait that forbids us to use trait objects for example for a more elegant solution. What I had to end up doing is a bunch of from-into functions to convert types between each other. Some of those restrictions came from uniffi.